src/internal_math.rs

```diff
@@ -0,0 +1,224 @@
+use std::mem::swap;
+
+mod atcoder {
+
+    mod internal {
+
+        /// # Arguments
+        /// * `m` `1 <= m`
+        ///
+        /// # Returns
+        /// x mod m
+        /* const */
+        fn safe_mod(mut x: i64, m: i64) -> i64 {
+            x %= m;
+            if x < 0 {
+                x += m;
+            }
+            x
+        }
+
+        /// Fast moduler by barrett reduction
```

Comment:
nit:
```suggestion
        /// Fast modular by barrett reduction
```

Reply:
The original document contains that typo, so I kept as it is :P
Of course we could fix it though.

Reply:
I think we don't want to have the same copy but want to have a library with the equivalent features as AtCoder's one. So if we find a bug in the original one, it's better to fix it.